packages/x-data-grid/src/tests/keyboard.DataGrid.test.tsx

```diff
@@ -62,7 +61,8 @@ describe('<DataGrid /> - Keyboard', () => {
           columnHeaderHeight={HEADER_HEIGHT}
           hideFooter
           filterModel={{ items: [{ field: 'id', operator: '>', value: 10 }] }}
-          experimentalFeatures={{ warnIfFocusStateIsNotSynced: true }}
+          // This had to be disabled again, `user.click` is not working with it
+          experimentalFeatures={{ warnIfFocusStateIsNotSynced: false }}
```

Comment:
I don't know why exactly this was causing issues. I suppose the warn check might be wrong? Or maybe it doesn't work properly with async? 🤔